core/sail/inferencer/src/main/java/org/eclipse/rdf4j/sail/inferencer/fc/FastRdfsForwardChainingSail.java

```diff
@@ -0,0 +1,209 @@
+package org.eclipse.rdf4j.sail.inferencer.fc;
+
+import org.eclipse.rdf4j.common.iteration.Iterations;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.eclipse.rdf4j.repository.Repository;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.eclipse.rdf4j.repository.RepositoryResult;
+import org.eclipse.rdf4j.sail.NotifyingSail;
+import org.eclipse.rdf4j.sail.SailException;
+import org.eclipse.rdf4j.sail.inferencer.InferencerConnection;
+import org.eclipse.rdf4j.sail.inferencer.fc.AbstractForwardChainingInferencer;
+
+import java.io.File;
+import java.util.*;
+
+/**
+ * @author Håvard Mikkelsen Ottestad
+ */
+
+public class FastRdfsForwardChainingSail extends AbstractForwardChainingInferencer {
```

Comment:
I would rather name it something else. "Fast" is somewhat subjective, and also not very descriptive as to _how_ this inferencer is different from the other inferencer. But I don't quite understand the base algorithm/design you're using here yet. Can you describe in more detail what strategy this inferencer uses and _why_ it is fast (or "faster than the standard inferencer", at least)?